src/main/scala/io/github/spark_redshift_community/spark/redshift/DefaultSource.scala

```diff
@@ -39,7 +37,7 @@ class DefaultSource(
   /**
    * Default constructor required by Data Source API
    */
-  def this() = this(DefaultJDBCWrapper, awsCredentials => new AmazonS3Client(awsCredentials))
+  def this() = this(DefaultJDBCWrapper, new AWSS3ClientFactory())
```

Comment:
Is this equivalent to what we have ? How are we passing over the awsCredentials object?

Reply:
I'm honestly not sure on this. I'm hoping someone with more Scala experience can help but it's my understanding that the original constructor was defining an anonymous function that would pass in the credentials into the constructor for AmazonS3Client. If the apply method is working then the AWSS3ClientFactory should work the same way. When I tested it SQL methods were working.